test/ApiRequestorTest.php

```diff
@@ -37,5 +37,52 @@ public function testEncodeObjects()
             ));
         }
     }
-    
+
+    /**
+     * @dataProvider provideValidAPITokens
+     *
+     * @param $expectedAuthorizationType
+     * @param $apiToken
+     * @throws ReflectionException
+     */
+    public function testGetAuthorizationType($expectedAuthorizationType, $apiToken)
+    {
+        // We have to do some work here because this is normally
+        // private. This is just for testing! Also it only works on PHP >=
+        // 5.3
+        if (version_compare(PHP_VERSION, '5.3.2', '>=')) {
```

Comment:
if having a private method is troublesome to test, I personally wouldn't be I opposed to extending the Public API surface area to have something like a `getRequestHeaders` to test against. Seems like a useful method for debugging too.

Reply:
Sure, I can refactor. I did things this way because the method above this also did things this way.